src/app/source-file-tabs/source-file-tabs.component.ts

```diff
@@ -72,28 +83,57 @@ export class SourceFileTabsComponent implements OnChanges {
             this.version.workflow_path,
             this.descriptorType
           );
-          if (this.fileTabs.size > 0) {
-            this.changeFileType(this.fileTabs.values().next().value);
-          }
           sourceFiles.forEach((sourceFile) => {
             if (this.isPrimaryDescriptor(sourceFile.path)) {
               this.primaryDescriptors.push(sourceFile);
               this.primaryDescriptorPath = sourceFile.path;
             }
           });
+          this.findFile();
         },
         () => {
           this.displayError = true;
         }
       );
   }
 
+  findFile() {
+    const queryFilePath = this.activatedRoute.snapshot.queryParams['file'];
+    if (this.fileTabs?.size > 0) {
+      // Attempt to find the file that is indicated by the 'file' query parameter.
+      // If we are successful, select the file, change the tab, and return.
+      if (queryFilePath) {
+        for (let tabIndex = 0; tabIndex < this.fileTabs.size; tabIndex++) {
+          const sourceFiles = Array.from(this.fileTabs.values())[tabIndex];
+          for (let sourceFile of sourceFiles) {
+            if (sourceFile.absolutePath === queryFilePath) {
+              this.selectFile(sourceFile);
+              this.changeTab(tabIndex);
+              return;
+            }
+          }
+        }
+      }
+      // Otherwise, select the first file in the first tab.
+      const files = this.fileTabs.values().next().value;
+      this.selectFile(files[0]);
```

Comment:
I don't think you can assume there will be at least one file. I believe several workflow versions without any source files.

Reply:
> I don't think you can assume there will be at least one file. I believe several workflow versions without any source files.

Believe it or not, in Typescript, that'll return `undefined` if there are zero elements.

Reply:
Ah, it does in JavaScript as well... I got confused with Java.